### PR TITLE
Make tutorial hint messages persistent

### PR DESCRIPTION
**Pull request by CelticMinstrel**
Based on @stevecotton's #6615, this moves many of the tutorial hint messages into persistent overlay text. I played through the tutorial with these changes and I think it seems pretty good, but it probably needs a little refinement.

### CODE REVIEW

**Comment by CelticMinstrel** [comment]
Here's an example of what it looks like.

**Comment by Wedge009** [comment]
I haven't tried this but as with #6615 in principle it sounds like a good idea. It probably addresses the concerns mentioned in #6614 where the instructional part of the dialogue is missed amongst the 'flavour' text that some players may skip over.

File: data/campaigns/tutorial/lua/hint_message.lua
Comment (CelticMinstrel) [reply data/campaigns/tutorial/lua/hint_message.lua]: Good point, thanks for spotting that.

**Comment by stevecotton** [comment]
Some of the message timings should change - for example, S01's `caption="Recruiting"` message currently displays when you move to the keep, but would fit better if it appeared at the start of the lesson, when the "I'll recruit some elves" message appears.

Some text is looking a bit cut-off (this was in a 1280x1024 Wesnoth window), only noticed it on this message.
cutoff_overlay_text

File: data/campaigns/tutorial/scenarios/02_Tutorial_part_2.cfg
Comment (stevecotton) [inline data/campaigns/tutorial/scenarios/02_Tutorial_part_2.cfg:293]: Why doesn't the schema validation trigger for the leftover `speaker` and `image` here?

Comment (CelticMinstrel) [reply data/campaigns/tutorial/scenarios/02_Tutorial_part_2.cfg:293]: The `[event]` tag is set to allow any child tag inside it. The effect is that the schema validates known ActionWML tags and ignores unknown ones. The reason it's defined this way is so that adding a custom tag doesn't automatically break the schema.

**Comment by Pentarctagon** [comment]
So aside from the width issue, this seems good to merge? The message timings seems unrelated to this PR's purpose though, since it's changing how a message is displayed rather than when.

**Comment by CelticMinstrel** [comment]
Yes, I don't know of anything to block merging this. The width issue would be an issue in the API, not in this change. It _would_ be nice to get a review from someone other than stevecotton though.

**Review by Vultraz — APPROVED** [review]

File: data/campaigns/tutorial/scenarios/02_Tutorial_part_2.cfg
Comment (stevecotton) [reply data/campaigns/tutorial/scenarios/02_Tutorial_part_2.cfg:293]: Still, they're leftovers that you should fix.

**Comment by Pentarctagon** [comment]
@CelticMinstrel So should I click merge, or were you planning to?

**Comment by CelticMinstrel** [comment]
Sure, I can do it.

**Pull request merged**
